lib/ejs.js

```diff
@@ -64,6 +65,18 @@ var _OPTS_PASSABLE_WITH_DATA = ['delimiter', 'scope', 'context', 'debug', 'compi
 // so we make an exception for `renderFile`
 var _OPTS_PASSABLE_WITH_DATA_EXPRESS = _OPTS_PASSABLE_WITH_DATA.concat('cache');
 var _BOM = /^\uFEFF/;
+var _JS_IDENTIFIER = /^[a-zA-Z_$][0-9a-zA-Z_$]*$/;
+
+var createObj = function() {
+  if (typeof Object.create !== 'function') {
+    return function (o) {
+      function F() {}
+      F.prototype = o;
+      return new F();
+    };
+  }
+  return Object.create;
+}();
```

Comment:
`new F()` doesn’t support `null` prototypes.

---

A `null`‑supporting `Object.create` polyfill created by me can be seen in the [proxy‑polyfill](https://github.com/GoogleChrome/proxy-polyfill/blob/f667dd4b25355b99a264e56d101307d50807e187/src/proxy.js#L34-L55) repository:
```js
const $Object = Object;

// Closure assumes that `{__proto__: null} instanceof Object` is always true, hence why we check against a different name.
const canCreateNullProtoObjects = Boolean($Object.create) || !({ __proto__: null } instanceof $Object);
const objectCreate =
	$Object.create ||
	(canCreateNullProtoObjects
		? function create(proto) {
			validateProto(proto);
			return { __proto__: proto };
		}
		: function create(proto) {
			validateProto(proto);
			if (proto === null) {
				throw new SyntaxError('Native Object.create is required to create objects with null prototype');
			}

			// nb. cast to convince Closure compiler that this is a constructor
			var T = /** @type {!Function} */ (function T() {});
			T.prototype = proto;
			return new T();
		});
```

Reply:
What happens in old runtimes in the case where the ctor's `prototype` property is set to `null`? Is it just a no-op? Dos your impl throw if it can't set the proto to `null`? Maybe we'd be better off with simple pass-through in that case.

Reply:
Thanks a ton for chiming in on this!

Reply:
> What happens in old runtimes in the case where the ctor's `prototype` property is set to `null`? Is it just a no-op? Dos your impl throw if it can't set the proto to `null`?

It creates an object with its prototype set to `%Object.prototype%` of the realm that `T` comes from:
- <https://www.ecma-international.org/wp-content/uploads/ECMA-262_3rd_edition_december_1999.pdf#sec-13.2.2>
- <https://tc39.es/ecma262/#sec-ecmascript-function-objects-construct-argumentslist-newtarget>

Reply:
So effectively a no-op. Well, damn. We were better off with the simple pass-through. Although I guess there's the middle case where `__proto__` is accessible. Since I've merged this, I'll just modify our `createObj`.

Reply:
Thanks again for chiming in, much appreciated!

Reply:
@ExE-Boss, can you take a quick look here? https://github.com/mde/ejs/pull/603